lib/linter/linter.js

```diff
@@ -1057,7 +1116,15 @@ function runRules(sourceCode, configuredRules, ruleMapper, parserName, languageO
         function addRuleErrorHandler(ruleListener) {
             return function ruleErrorHandler(...listenerArgs) {
                 try {
-                    return ruleListener(...listenerArgs);
+                    const ruleListenerReturn = ruleListener(...listenerArgs);
+
+                    const ruleListenerResult = stats ? ruleListenerReturn.result : ruleListenerReturn;
+
+                    if (stats) {
+                        storeTime(ruleListenerReturn.tdiff, { filename: sharedTraversalContext.getPhysicalFilename(), type: "rules", key: ruleId }, slots, slots.resetTimes === filename);
```

Comment:
Perhaps `slots.resetTimes === filename` is why the numbers are lower than expected, looks like each listener call resets time?

Reply:
Since the stats are now stored in the Linter, maybe Linter doesn't need to track them per filename, but only for the last run? That way, stats could not accumulate for multiple runs on the same file.

There are already methods that return data from the last run only, like `Linter#getSuppressedMessages()` and the newly added `Linter#getFixPasses()`.

So, `Linter#getStats()` would be returning stats from the last run. The stats would be reset on a new run, that is when `Linter#verify()` or `Linter#verifyAndFix()` is called.

Reply:
@mdjermanovic: I just changed the `resetTimes` conditions. Now, if a file is parsed *again* the rules timing for that file will be reset. 👍 The *rules* output is now as expected and matches that of TIMING:
```
TIMING=all node bin/eslint tests/lib/linter/linter.js --stats -f json
[{"filePath":"/Users/I534083/Public/eslint-fork/tests/lib/linter/linter.js",
...
"stats":{"times":{"passes":[{
  "parse":{"total":179.162125},
  "rules":{"indent":{"total":140.72399700002433}},
  "fix":{"total":0},"total":402.135417}]},
  "fixPasses":0},
...
}]
Rule   | Time (ms) | Relative
:------|----------:|--------:
indent |   140.724 |   100.0%
```

Reply:
I still see the same version, maybe you didn't push the changes?

Reply:
I just pushed them 👍 Please have a look now.

Reply:
> I think we should simply reset stats at the start of verify/verifyAndFix, and I also think there's no need to keep stats per filename in Linter.

:+1:

Reply:
@mdjermanovic: I agree, this also simplifies things nicely! 👍 I've just updated the code - could you please have another look?